### DISCUSSION TITLE
Checking about use-case

### DISCUSSION THREAD

**Discussion by galleon** [casdoor/casdoor#1677] [Q&A] [answered]
Hi there

Just discovered casdoor. It looks great but before adopting it (probably SAAS), i would like to check my use-case. Here is what I would like to use it for:
  - Authn for a multi-tenant app
  - Fine grain authz (object level of a Django app)
  - Authz of external services (AWS Lambda)
  Happy to hear your feedback on those points ?
  
 Thanks

**Comment by nomeguy** [comment, marked as answer]
Hi @galleon glad to answer your questions:

> Authn for a multi-tenant app

Casdoor supports multi-tenancy natively via organizations. One organization is a tenant: https://casdoor.org/docs/organization/overview

> Fine grain authz (object level of a Django app)

Casdoor uses Casbin as authorization system. Casbin has well support for Django ecosystem, like:

- https://github.com/pycasbin/django-casbin
- https://github.com/pycasbin/django-orm-adapter

> Authz of external services (AWS Lambda)

Casdoor has exposed Casbin Enforcement API via HTTP RESTful, so you can use Casdoor as an centralized authorization server and let your AWS Lambda code call into Casdoor via HTTP requests for authorization decision: https://casdoor.org/docs/permission/exposed-casbin-apis
